sx127x/sx127x.go

```diff
@@ -119,9 +130,24 @@ func (d *Device) SetOpMode(mode uint8) {
 	d.WriteRegister(SX127X_REG_OP_MODE, new)
 }
 
-// SetOpMode changes the sx1276 mode
+// SetOpModeLora changes the sx1276 mode to lora.
 func (d *Device) SetOpModeLora() {
-	d.WriteRegister(SX127X_REG_OP_MODE, SX127X_OPMODE_LORA)
+	d.WriteRegister(SX127X_REG_OP_MODE, d.ReadRegister(SX127X_REG_OP_MODE)|SX127X_OPMODE_LORA)
+}
+
+// SetOpModeFsk changes the sx1276 mode to fsk/ook.
+func (d *Device) SetOpModeFsk() {
+	d.WriteRegister(SX127X_REG_OP_MODE, d.ReadRegister(SX127X_REG_OP_MODE)&^SX127X_OPMODE_LORA)
+}
+
+// SetModulationType changes the modulation type (0=FSK, 1=OOK)
+func (d *Device) SetModulationType(typ uint8) {
+	switch typ {
+	case SX127X_OPMODE_MODULATION_FSK:
+		d.WriteRegister(SX127X_REG_OP_MODE, d.ReadRegister(SX127X_REG_OP_MODE)&^(1<<SX127X_OPMODE_MODULATION_Pos))
+	case SX127X_OPMODE_MODULATION_OOK:
+		d.WriteRegister(SX127X_REG_OP_MODE, d.ReadRegister(SX127X_REG_OP_MODE)|(1<<SX127X_OPMODE_MODULATION_Pos))
+	}
 }
```

Comment:
Documentation says this is two bit field, although only 0b00 and 0b01 are used, it would maybe be safer to use mask like:
```suggestion
const (
	SX127X_OPMODE_MODULATION_MASK = uint8(0x60)
	SX127X_OPMODE_MODULATION_FSK = uint8(0x0)
	SX127X_OPMODE_MODULATION_OOK = uint8(0x20)
)
func (d *Device) SetModulationType(typ uint8) {
	cleared := d.ReadRegister(SX127X_REG_OP_MODE)&^SX127X_OPMODE_MODULATION_MASK
	d.WriteRegister(SX127X_REG_OP_MODE, cleared|typ)
}
```